tools/set_version.py

```diff
@@ -49,6 +49,8 @@ def substitute_file(name):
             line = '    version="%d.%d.%d",\n' % (version[0], version[1], version[2])
         elif '"-LT' in line and name.endswith('settings_pack.cpp'):
             line = re.sub('"-LT[0-9A-Za-z]{4}-"', '"-LT%c%c%c%c-"' % v(version), line)
+        elif 'local FULL_VERSION = ' in line and name == 'Jamfile':
+            line = '\tlocal FULL_VERSION = %d.%d.%d ;\n' % (version[0], version[1], version[2])
```

Comment:
Actually, I probably answered my own question, this didn't work with v2.0.1 for me:

```
error: patch failed: tools/set_version.py:49
error: tools/set_version.py: patch does not apply
```

Reply:
it's likely some additional patches on top of `v1.2.12` (that are in `RC_1_2`) prevent this part of the patch from applying. This is not important though